enterprise/backend/src/metabase_enterprise/sandbox/query_processor/middleware/row_level_restrictions.clj

```diff
@@ -301,38 +303,54 @@
         (_ :guard (every-pred map? :source-table))
         (assoc &match ::gtap? true)))))
 
+(defn- expected-cols [query]
+  (binding [*current-user-permissions-set* (atom #{"/"})]
+    ((requiring-resolve 'metabase.query-processor/query->expected-cols) query)))
```

Comment:
Just a note, should we use a notion of the `classloader/require` here versus "plain" require?

Reply:
Yeah good call